core/variable_binary_arithmetic.cpp

```diff
@@ -279,4 +304,30 @@ Variable operator/(const double a, const VariableConstProxy &b_proxy) {
   return b;
 }
 
+struct MakeVariableWithType {
+  template <class T> struct Maker {
+    static Variable apply(const VariableConstProxy &parent) {
+      return transform<double, float, int64_t, int32_t, bool>(
+          parent, overloaded{[](const units::Unit &x) { return x; },
+                             [](const auto &x) {
+                               if constexpr (detail::is_ValueAndVariance_v<
+                                                 std::decay_t<decltype(x)>>)
+                                 return detail::ValueAndVariance<T>{
+                                     static_cast<T>(x.value),
+                                     static_cast<T>(x.variance)};
+                               else
+                                 return static_cast<T>(x);
+                             }});
+    }
+  };
+  static Variable make(const VariableConstProxy &var, DType type) {
+    return CallDType<double, float, int64_t, int32_t, bool>::apply<Maker>(type,
+                                                                          var);
+  }
```

Comment:
Why is this wrapping function required? And why do we need the `struct MakeVariableWithType`?

Reply:
To keep the corresponding functionality grouped in one instance I would add the types list for both `transform` and `CallDType` as
```c++
using types = std::tuple<....>;
```
they can't work with tuples.